lib/api/filter.rb

```diff
@@ -120,24 +122,16 @@ def single_expression(field, operator, value)
     end
 
     def split_filter_string(filter)
-      operator = nil
-      operators_from_longest_to_shortest = OPERATORS.keys.sort_by(&:size).reverse
-      filter.size.times do |i|
-        operator = operators_from_longest_to_shortest.detect do |o|
-          o == filter[(i..(i + o.size - 1))]
-        end
-        break if operator
-      end
+      filter_match = OPERATOR_REGEXP.match(filter)
+      filter_attr, operator, filter_value = filter_match[1..3] if filter_match
 
-      if operator.blank?
+      if operator.blank? || !OPERATORS.key?(operator)
```

Comment:
Since a blank operator wouldn't be in the list anyway, can you do:

```suggestion
      unless OPERATORS.key?(operator)
```